bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/BlueZBluetoothDevice.java

```diff
@@ -247,6 +256,10 @@ public boolean readCharacteristic(BluetoothCharacteristic characteristic) {
             throw new IllegalStateException("TinyB device is not yet set");
         }
         BluetoothGattCharacteristic c = getTinybCharacteristicByUUID(characteristic.getUuid().toString());
+        if (c == null) {
+            logger.warn("Characteristic '{}' is missing on device '{}'.", characteristic.getUuid(), address);
```

Comment:
Warnings in logs should be shown, if there is a bug situation that the user should report - for doing so, the stacktrace should be added.
If this is rather an "expected" case and we know that we sometimes ask for characteristics that are not present, the logging should be changed to debug.

Reply:
This would be an unexpected case where somehow after service discovery the underlying tinyb characteristics go missing, which frankly should never happen, but I've seen the underlying device disappear from bluez before, which I also think should never happen, yet it does. But I guess you could also trigger this if you try to use a BluetoothCharacteristic from an entirely different device, but it would be really hard to pull that off with the existing framework.

Reply:
Also, these warning messages were added to to match the behavior of the `enableNotifications(...)` and `disableNotifications(...)` methods, which also log a missing characteristic at the warn logging level.